### PR TITLE
Session-scoped authentication

### PR DESCRIPTION
**Pull request by stefano-ottolenghi**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-drivers-443.surge.sh

Updated pages:
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/go-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/go-manual/5/transactions)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/java-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/java-manual/5/transactions)
- [Connection](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/connect)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/transactions)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/python-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/python-manual/5/transactions)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-drivers-443.surge.sh

Updated pages:
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/go-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/go-manual/5/transactions)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/java-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/java-manual/5/transactions)
- [Connection](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/connect)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/transactions)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/python-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/python-manual/5/transactions)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-drivers-443.surge.sh

Updated pages:
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/go-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/go-manual/5/transactions)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/java-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/java-manual/5/transactions)
- [Connection](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/connect)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/transactions)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/python-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/python-manual/5/transactions)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-drivers-443.surge.sh

Updated pages:
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/go-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/go-manual/5/transactions)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/java-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/java-manual/5/transactions)
- [Connection](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/connect)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/transactions)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/python-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/python-manual/5/transactions)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-drivers-443.surge.sh

Updated pages:
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/go-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/go-manual/5/transactions)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/java-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/java-manual/5/transactions)
- [Connection](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/connect)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/transactions)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/python-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/python-manual/5/transactions)

File: go-manual/modules/ROOT/pages/query-simple.adoc [hunk 1]
```diff
@@ -227,24 +227,37 @@ In other words, there is no guarantee that a write query submitted in read mode
...
+[role=label--new-5.14]
 === Run queries as a different user
 
-You can execute a query under the security context of a different user with the callback `neo4j.ExecuteQueryWithImpersonatedUser("<somebodyElse>")`, specifying the name of the user to impersonate.
-For this to work, the user under which the `DriverWithContext` object
-was created needs to have the link:{neo4j-docs-base-uri}/cypher-manual/current/administration/access-control/dbms-administration#access-control-dbms-administration-impersonation[appropriate permissions].
-Impersonating a user is cheaper than creating a new `DriverWithContext` object.
+You can execute a query through a different user with the configuration callback `neo4j.ExecuteQueryWithAuthToken()`.
+Switching user at the query level is cheaper than creating a new `DriverWithContext` object.
+The query is then run within the security context of the given user, and not the one under which the driver was instantiated with (i.e., home database, permissions, etc.). +
```
Comment (robsdedude) [inline go-manual/modules/ROOT/pages/query-simple.adoc]: The second half seems a little odd. Is there an extra "under" or "with" or such?

File: go-manual/modules/ROOT/pages/query-simple.adoc [hunk 2]
```diff
@@ -227,24 +227,37 @@ In other words, there is no guarantee that a write query submitted in read mode
...
     "MATCH (p:Person) RETURN p.name",
     nil,
     neo4j.EagerResultTransformer,
     neo4j.ExecuteQueryWithDatabase("neo4j"),
-    neo4j.ExecuteQueryWithImpersonatedUser("<somebodyElse>"))
+    neo4j.ExecuteQueryWithAuthToken(queryAuth))
 ----
 
-When impersonating a user, the query is run within the complete security context of the impersonated user and not the authenticated user (i.e. home database, permissions, etc.).
+The callback `neo4j.ExecuteQueryWithImpersonatedUser()` provides a similar functionality, is available earlier than version 5.14, and works on all server versions.
```
Comment (robsdedude) [inline go-manual/modules/ROOT/pages/query-simple.adoc]: To be precise, impersonation was introduced in v4.4 (both server and driver)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-drivers-443.surge.sh

Updated pages:
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/go-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/go-manual/5/transactions)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/java-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/java-manual/5/transactions)
- [Connection](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/connect)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/javascript-manual/5/transactions)
- [Query the database](https://neo4j-docs-drivers-443.surge.sh/python-manual/5/query-simple)
- [Run your own transactions](https://neo4j-docs-drivers-443.surge.sh/python-manual/5/transactions)

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

**Review by MaxAake — APPROVED** [review]
Javascript looks good to me!

**Pull request merged**